packages/light-client-extension-helpers/src/smoldot/index.ts

```diff
@@ -0,0 +1,3 @@
+export * from "./client"
+export * from "./types"
+export * from "./tasks"
```

Comment:
I know that this pattern exists throughout the library, but IMO I think it'd be better practice to be explicit about the exports from a package rather than use * exports (just so the library interface is obvious just looking at this root file and not needing to traverse other stuff :)

Reply:
Yeah but I find its really easy to forget to add a new type that should get re-exported and then it gets missed during code review because people don't really pay attention to that. 

For anything that should be kept 100% internal I like to use the `@internal` jsdoc tag and then with `stripeInternal` set `true` in your tsconfig, these definitions won't be included in the d.ts`

https://www.typescriptlang.org/tsconfig/#stripInternal

Reply:
Yup totally true; we forget to export types in Rust a fair bit, but personally I much prefer to forget exporting a type than to accidentally export a type I didn't mean to; it's easier to go one way than the other!

TIL about `@internal`; that's a cool approach if used thoroughly :)

Reply:
Gonna opt for the YOLO approach and just use @internal.

But if you want to actually see what the publicly export types are u can run `corepack pnpm turbo build` then open the `dist` folder and each of the the subpath exports will have a `d.ts` file in it showing the public api.

![image](https://github.com/paritytech/substrate-connect/assets/21375952/27fc33f4-d20f-42a9-bc3c-f72cf88e58e5)